syntax = "proto3";

package engine;

//import "google/protobuf/empty.proto"; 
//import "google/protobuf/any.proto";
 
service Engine{
    rpc IsAlive(IsAliveRequest) returns (IsAliveResponse) {}
    rpc Status(StatusRequest) returns (StatusResponse) {}
    rpc Config(ConfigRequest) returns (ConfigResponse) {}
}

    // ENGINE //

message IsAliveRequest{}

message IsAliveResponse{
    bool is_alive = 1;
}

message StatusRequest{}

message StatusResponse{
    bool is_alive = 1;
    bool background_is_alive = 2;
    bool broom_thread_is_alive = 3;
    bool changes_domains_thread_is_alive = 4;
    bool changes_hyps_thread_is_alive = 5;
    bool download_changes_thread_is_alive = 6;
    bool event_thread_is_alive = 7;
    repeated string disk_operations_threads = 8;
    repeated string long_operations_threads = 9;
    repeated string working_threads = 10;
    repeated string status_threads = 11;
}


message ConfigRequest{
    ConfigIntervals intervals = 1;
    ConfigSsh ssh = 2;
    ConfigStats stats = 3;
    ConfigLog log = 4;
    ConfigTimeouts timeouts = 5;
}

message ConfigResponse{
    ConfigIntervals intervals = 1;
    ConfigSsh ssh = 2;
    ConfigStats stats = 3;
    ConfigLog log = 4;
    ConfigTimeouts timeouts = 5;
}





message ConfigIntervals{
    int32 status_polling = 1;
    int32 time_between_polling = 2;
    int32 test_hyp_fail = 3;
    int32 background_polling = 4;
    int32 transitional_states_polling = 5;
}
message ConfigSsh{
    bool paramiko_host_key_policy_check = 1;
}
message ConfigStats{
    bool active = 1;
    int32 max_queue_domains_status = 2;
    int32 max_queue_hyps_status = 3;
    int32 hyp_stats_interval = 4;
}
message ConfigLog{
    string log_name = 1;
    string log_level = 2;
    string log_file = 3;
}
message ConfigTimeouts{
    int32 ssh_paramiko_hyp_test_connection = 1;
    int32 timeout_trying_ssh = 2;
    int32 timeout_trying_hyp_and_ssh = 3;
    int32 timeout_queues = 4;
    int32 timeout_hypervisor = 5;
    int32 libvirt_hypervisor_timeout_connection = 6;
    int32 timeout_between_retries_hyp_is_alive = 7;
    int32 retries_hyp_is_alive = 8;
}
